### PR TITLE
Add JAX version of `legacy.utils.random_single_categorical`

### PR DESCRIPTION
**Pull request by conorheins**
To address #304 (and drive-by addressing of #308)

### Summary
As part of the JAX backend integration, this is one PR of a push to replace all use of `legacy.utils` with the newer `jax.numpy` based utility functions. This PR adds `random_factorized_categorical` to `pymdp.utils` as a JAX-based replacement for `random_single_categorical` and replaces the usage of the old numpy version in the repo and unit tests. The function generates a random factorized categorical distribution (represented as a list of marginal distributions, stored as `(dim[f],)` sized 1D ndarrays) using jax.random.PRNGKey control.

#### Drive-by fixes
- factor out `test_utils.py` into `test_utils.py` (now exclusively for testing the old legacy utilities) and `test_utils_jax.py` (for the new JAX based utilities)
- add jax version of `norm_dist_obj_array` which just tree-maps `norm_dist` over a collection of jax.numpy arrays, addressing #308

#### Note
Moving to draft until #310 and #311 are merged

### CODE REVIEW

File: pymdp/utils.py [hunk 1]
```diff
@@ -46,6 +50,18 @@ def validate_normalization(tensor: Tensor, axis: int = 1, tensor_name: str = "te
...
+def random_factorized_categorical(key, dims_per_var: Sequence[int]) -> List[jax.Array]:
+    """"
+    Creates a list of jax arrays representing random Categorical distributions with dimensions
+    given by dims_per_var[i]. In the context of observations or hidden state posteriors,
+    this can seen as a factorized categorical distribution over multiple variables, i.e.
+    P(X1, X2, ..., Xn) = P(X1)P(X2)...P(Xn)
+    """
+
+    num_vars = len(dims_per_var)
+    keys = jax.random.split(key, num_vars)
```
Comment (tverbele) [inline pymdp/utils.py]: use jr.split?

**Review by tverbele — APPROVED** [review]
lgtm ... one instance where you use jax.random whereas you import jax.random as jr as well, just for consistency

File: pymdp/utils.py (hunk 1 above)
Comment (conorheins) [reply pymdp/utils.py]: eagle eye! thanks for spotting

Comment (conorheins) [inline pymdp/utils.py]:
```suggestion
    keys = jr.split(key, num_vars)
```
